c2cpg/src/main/scala/io/shiftleft/c2cpg/parser/HeaderFileFinder.scala

```diff
@@ -0,0 +1,39 @@
+package io.shiftleft.c2cpg.parser
+
+import better.files._
+import org.jline.utils.Levenshtein
+
+import java.nio.file.{FileSystems, Path}
+
+class HeaderFileFinder(roots: List[String]) {
+
+  private val headerExtensions = List(".h", ".hpp", ".hh")
+  private val nameToPathMap: Map[String, List[Path]] = {
+    roots
+      .flatMap { root =>
+        File(root).listRecursively.filter(f => f.extension.isDefined && headerExtensions.contains(f.extension.get))
```

Comment:
It might be quite slow doing this.
We could also reuse this: https://github.com/ShiftLeftSecurity/codepropertygraph/blob/master/c2cpg/src/main/scala/io/shiftleft/c2cpg/C2Cpg.scala#L42

Reply:
You're right, I've written a better version before ;) - Reusing `SourceFiles.determine`